builtin/provisioners/remote-exec/resource_provisioner.go

```diff
@@ -272,7 +272,7 @@ func retryFunc(ctx context.Context, timeout time.Duration, f func() error) error
 			}
 
 			// Try the function call
-			err := f()
+			err := backoff.Retry(f, backoff.NewExponentialBackOff())
```

Comment:
`backoff.Retry` already retries the function until there's a nil error, so this needs to be pulled out of the for loop.